### PR TITLE
88 handle empty slc list

### LINKED ISSUE #88: from_binary produces a confusing error when slc_files is empty
If `slc_files` is an empty iterable, `from_binary` silently skips the read loop, `slcs` stays `None`, and the eventual `ds.assign({vlabel: (("azimuth","range","time"), None)})` surfaces as:

```
ValueError: Variable 'complex': Could not convert tuple of form
(dims, data[, attrs, encoding]): (('azimuth', 'range', 'time'), None) to Variable.
```

It took me a while to trace this back to "no files were found". This is easy to hit when the user runs the tutorial notebook before downloading the dataset — `path.rglob('*.raw')` returns `[]` and the failure mode is opaque.

An early guard would make the failure obvious:

```python
if len(slc_files) == 0:
    raise ValueError("`slc_files` is empty; no SLC files to read.")
```

(Filed in the context of JOSS review openjournals/joss-reviews#10492.)

### PR DESCRIPTION
**Pull request by rogerkuou**
fix #88 

handled empty slc list
added more test for edge cases, mainly for metadata reading.

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @Simon-van-Diepen this is the second PR for a JOSS feedback. mainly working on handling edge cases in `_io.py`

File: sarxarray/_io.py [hunk 1]
```diff
@@ -416,18 +425,11 @@ def _parse_metadata(file, driver, ifg_file_name):
             with open(file_ifg) as f_ifg:
                 content_ifg = f_ifg.read()
                 for key, pattern in RE_PATTERNS_DORIS5_IFG.items():
-                    if key in META_ARRAY_KEYS.keys():  # multiple hits allowed
```
Comment (Simon-van-Diepen) [inline sarxarray/_io.py:419]: I don't understand the removal of the `re.findall` option here. If I understand it correctly `re.search` will only return the first hit, whereas the keys in `META_ARRAY_KEYS` are specifically designed to have multiple hits and return arrays of data. I think we need to re-add this to prevent breaking of the orbit information reading

File: sarxarray/_io.py [hunk 2]
```diff
@@ -447,16 +449,8 @@ def _regulate_metadata(metadata, driver):
...
         time_format = TIME_FORMAT_DORIS4
         unit_conversions = META_UNIT_CONVERSION_MULTIPLICATION_KEYS_DORIS4
-    else:
-        raise NotImplementedError(
-            f"Driver '{driver}' is not implemented. "
-            "Supported drivers are: 'doris4', 'doris5'."
-        )
 
     list_time = []
-    # If the time is a single string, convert it to a list
```
Comment (Simon-van-Diepen) [inline sarxarray/_io.py:457]: I don't understand the removal of this part either. Is this an edge case that is impossible to achieve?

**Review by Simon-van-Diepen — CHANGES_REQUESTED** [review]
Hi Ou,
I am uncertain about the removal of two parts of the code. Let's discuss

File: sarxarray/_io.py (hunk 1 above)
Comment (rogerkuou) [reply sarxarray/_io.py:419]: Hi @Simon-van-Diepen , I removed this because IMO it's never got hit because this part is about Doris v5 ifg metadata files (`ifg.res`). If I understand correctly, the keys in `META_ARRAY_KEYS` meant to handle the matrix-like metadata, which only exists in `metadata.res`. 

Parsing `META_ARRAY_KEYS` in `metadata.res` is already handled by [this part](https://github.com/TUDelftGeodesy/sarxarray/blob/c47817e0f677dbac3c8c5610015b0574b4561641/sarxarray/_io.py#L398-L403).

If you find this differently, could you please point me to an example `ifg.res` which contains matrix like metadata? I can add it to the unittest.

File: sarxarray/_io.py (hunk 2 above)
Comment (rogerkuou) [reply sarxarray/_io.py:457]: Hi @Simon-van-Diepen , I think yes, this case is impossible to achieve because of [these lines](https://github.com/TUDelftGeodesy/sarxarray/blob/c47817e0f677dbac3c8c5610015b0574b4561641/sarxarray/_io.py#L364-L368)

We initiated metadata from a default dict of list, and only append new values to each field. This is already a quite clean way to enforce the list metada at this stage (and later depends on if it;s a single value list we unfold again). So I think this part is not needed,

File: sarxarray/_io.py (hunk 1 above)
Comment (Simon-van-Diepen) [reply sarxarray/_io.py:419]: I completely missed that this is under the ifg section. You are correct, there is no key there and it should be handled by the part you linked.

File: sarxarray/_io.py (hunk 2 above)
Comment (Simon-van-Diepen) [reply sarxarray/_io.py:457]: This one I also did not spot, you are correct, this is indeed impossible to achieve

**Review by Simon-van-Diepen — APPROVED** [review]
After clarification I understand all these changes and they are good to go
